Dockerfile

```diff
@@ -29,27 +30,56 @@ RUN apt-get update -y \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+RUN bundle config --global silence_root_warning 1
+RUN bundle config --global path $BUNDLE_TO
+RUN bundle config --global frozen true
+
+# Setup service
+COPY --link playbook-website/services/puma.sh /home/app/src/playbook-website/services/puma.sh
+RUN chmod +x /home/app/src/playbook-website/services/puma.sh
+RUN mkdir /etc/service/puma && ln -s /home/app/src/playbook-website/services/puma.sh /etc/service/puma/run
 WORKDIR /home/app/src
 
-# Build Library
-COPY playbook-website/package.json playbook-website/
-COPY playbook/package.json playbook/
-COPY package.json .rubocop.yml .eslintrc.json .yarnrc.yml yarn.lock .npmrc ./
-COPY .yarn ./.yarn
-RUN --mount=type=secret,id=yarnenv,required env $(cat /run/secrets/yarnenv | xargs) yarn install
-RUN curl https://github.com/sass/node-sass/releases/download/v4.13.0/linux-x64-64_binding.node -o node_modules/node-sass/vendor/linux-x64-64_binding.node
+FROM base as rubypackages
+RUN --mount=type=bind,target=/tmp/src \
+    cd /tmp/src/ && \
+    find . -mindepth 0 -maxdepth 6 \( -name 'Gemfile' -o -name 'Gemfile.lock' -o -name '*.gemspec' \) -exec cp --parents "{}" /home/app/src \;
```

Comment:
@TeamTeaTime what is the reason to create a bind mount and copy only the files necessary to build?